Contest_API.md

```diff
@@ -1613,9 +1649,10 @@ Each problem object within the scoreboard consists of:
 
 #### Access restrictions at WF
 
-For clients with the `public` role, the scoreboard will not include
-information from judgements of submissions received after the scoreboard
-has been frozen until it has been thawed.
+For clients with the `public` or `team` role, the scoreboard will not
+include teams that are in any group that is hidden, nor information
```

Comment:
Just a question: what if a team is part of both visible and hidden groups? I guess there are arguments both for making such a team visible or not. Was this discussed?
Is it is now, it's easy to make a team hidden, which is convenient e.g. for jury/admin associated teams. OTOH, it makes it difficult to assign some hidden groups that might only be visible to admins/jury.

Reply:
We didn't discuss this, and I wasn't initially planning on addressing it. When doing the rest I noticed that we didn't specify what to do if a team is in both non-hidden and hidden groups, nor whether a team should see their own solved runs in the scoreboard during freeze. I had a short chat with Nicky comparing what CDS and DOMjudge do (surprise, it wasn't consistent) and decided the above was the most consistent way to address this.

To do what you're suggesting I think we would need to separate whether a group is hidden from whether it hides teams on the scoreboard, which means a new (likely optional) attribute. If you think it's a case we should support then I think we should discuss on the next call.

Reply:
There are two different concepts here. Hiding all the members of the group or hiding the group itself. The meaning of `hidden` in groups is (and has always been) the former. That said, the latter would definitely be more natural and make more sense. Since the meaning is "hide the members of the group", IMO the only reasonable interpretation is that you are hidden from the (main) scoreboard if you are a member of *any* hidden group, regardless of whether you are a member of any non-hidden groups.

All that said, I have never liked how we use `hidden`. I consider it a hack to allow for a commonly used feature (in DOMjudge) where you can have not only subset scoreboards (which is what groups allow for) but also superset  scoreboards (i.e. there are scoreboards that contain more teams then the default scoreboard, i.e. the hidden teams). 

I think a nicer way to support this could be to have a "default" group. At most one group may be the default group, and the teams in the default group is what you get if you query `/scoreboard` without a group_id argument (i.e. if `allteams` is the default group then `/scoreboard` and `/scoreboard?group_id=allteams` returns the exact same scoreboard). Furthermore, if no group is explicitly set as default, then the implicit group "all the teams" is default (otherwise you would almost always need to add an extra group of all teams just to make it the default).

Taking this even further (and maybe too far?). We do have contests where there is no scoreboard of all the teams. Specifically the NAQ and NADC. For these contests you *could* define an empty group to be the default group. This would cause `/scoreboard` to return an empty scoreboard, and you would *have* to specify a group_id to get anything useful. (It would kinda break the concept of local vs global rank referred ti in the group scoreboard section, but t.b.h. so does having any hidden team already.